Beginners/Lesson Three/README.md

```diff
@@ -0,0 +1,184 @@
+
+Knowing what to put in a functions parameters is basically a skill you learn over the years, but as I've done it for years we'll skip to what I want. I know that both of those bits of information can be found on an Artwork. So rather than providing two parameters, we could provide just one. Less, in general, is better. There's nuance there though. To do this we need to include a date string on the Artwork Type.
+
+``` swift
+struct Artwork {
+    var name: String
+    var medium: String
+    var availability: ArtworkAvailablility
+    var date: String
+}
+```
+
+OK. That's a date. Let's change our function to take any instance of an Artwork.
+
+``` swift
+func printArtworkSubtitle(artwork: Artwork) {
+    print(artwork.name)
+    print(artwork.date)
+}
+```
+
+Now, we can create an Artwork. We're going to start trying to represent real data. So we're gonna use my favourite artist, [Glenn Brown](https://www.artsy.net/artist/glenn-brown)
+
+``` swift 
+// https://www.artsy.net/artwork/glenn-brown-death-disco
+
+var disco = Artwork(name: "Death Disco", medium: "Oil on Panel", availability: .NotForSale, date: "2004")
+```
+
+OK, great. So let's print the subtitle for the artwork.
+
+``` swift
+printArtworkSubtitle(disco)
+```
+
+Let's recap. We've created a function that has an input of an Artwork type. This function looks inside the artwork to print out it's name, and it's date. This means whenever we want to print an artworks subtitle we can do so really easy, and very importantly, consistently. 
+
+### WhOOPs
+
+So let's get back to our `struct` Artwork. We were talking about how it contains variables. Well, they can also contain functions too. From my perspective, it is the job of the Artwork to decribe itself. So we're going to move the `printArtworkSubtitle` function into the Artwork struct.
+
+``` swift
+struct Artwork {
+    var name: String
+    var medium: String
+    var availability: ArtworkAvailablility
+    var date: String
+
+    func printArtworkSubtitle(artwork: Artwork) {
+        print(artwork.name)
+        print(artwork.date)
+    }
+}
+```
+
+done. Ish. This still works fine. We can see this by doing:
+
+```
+disco.printArtworkSubtitle(disco)
+```
+
+However Types come with a sneaky variable that can be used to reduce the amount of code we have to write. This variable is called `self`, and when you are using it, it refers to the _instance_ of which the function is attached. So instead of passing in an artwork, we can use the one that comes in implicitly.
```

Comment:
Technically, we don't need to refer to `self` in the function (unlike ObjC), so it might make sense to gloss over the distinction.